ui/lib/src/elements/where-space-dialog.ts

```diff
@@ -0,0 +1,101 @@
+import { html, css, LitElement } from "lit";
+import { property } from "lit/decorators.js";
+
+import { sharedStyles } from "../sharedStyles";
+import { contextProvided } from "@lit-labs/context";
+import { ScopedElementsMixin } from "@open-wc/scoped-elements";
+import { WhereStore } from "../where.store";
+import { whereContext, Space } from "../types";
+import { Dialog, TextField, Button, Checkbox, Formfield } from "@scoped-elements/material-web";
+
+/**
+ * @element where-space
+ */
+export class WhereSpaceDialog extends ScopedElementsMixin(LitElement) {
+  @property({ type: String }) current = "";
+
+  /** Dependencies */
+  @contextProvided({ context: whereContext })
+  _store!: WhereStore;
+
+  open() {
+    const dialog = this.shadowRoot!.getElementById("space-dialog") as Dialog
+    dialog.open = true
+  }
+
+  private async handleSpaceDialog(e: any) {
+    const name = this.shadowRoot!.getElementById(
+      "space-dialog-name"
+    ) as TextField;
+    const url = this.shadowRoot!.getElementById(
+      "space-dialog-url"
+    ) as TextField;
+    const multi = this.shadowRoot!.getElementById(
+      "space-dialog-multi"
+    ) as Checkbox;
+    if (e.detail.action == "ok") {
+      const img = this.shadowRoot!.getElementById("sfc") as HTMLImageElement;
+      img.onload = async () => {
+        const space: Space = {
+          name: name.value,
+          surface: {
+            url: url.value,
+            size: { x: img.naturalHeight, y: img.naturalWidth },
+            data: "[]",
+          },
+          meta: {
+            multi: multi.checked ? "true" : "",
+          },
+          wheres: [],
+        };
+        this.current = await this._store.addSpace(space);
```

Comment:
I assume this is the place in which you two way data binding. I would say this is actually discouraged here, if we want to make `<where-space-dialog>` reusable and not depend on the `<where-controller>` to be its parent (if one depends on the other to be its immediate parent, then maybe they shouldn't be separate components, since their reusability is actually little).

So, instead of passing the current property and update it (which actually `<where-space-dialog>` doesn't need to render at all if I'm seeing correctly), I would dispatch an event when the space is added, [docs](https://lit.dev/docs/components/events/#dispatching-events).

```ts
        this.current = await this._store.addSpace(space);
```

turns into 

```ts
        const newSpace = await this._store.addSpace(space);
        this.dispatchEvent(new CustomEvent('space-added', { detail: newSpace, bubbles: true, composed: true });
```

and then in the `<where-controller>`:

```ts
<where-space-dialog id="space-dialog" @space-added=${e => this.current = e.detail}></where-space-dialog>
```

Is this making sense?